File: src/main/java/controller/TimerAction.java
package controller;

import model.SnakeDeadException;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class TimerAction implements ActionListener {

    private boolean stop = false;

    private final SnakeController snakeController;

    public TimerAction(SnakeController snakeController) {
        this.snakeController = snakeController;
    }

    public void actionPerformed(ActionEvent e) {
        try {
            if (!stop) {
                snakeController.step();
            }
        } catch (SnakeDeadException ex) {
            stop = true;
            snakeController.stop();
        }
    }

}

File: src/main/kotlin/co/electric/snake/controller/TimerAction.kt
package co.electric.snake.controller

import co.electric.snake.model.SnakeDeadException
import java.awt.event.ActionEvent
import java.awt.event.ActionListener

class TimerAction(private val snakeController: SnakeController) : ActionListener {

    private var stop = false

    override fun actionPerformed(event: ActionEvent) {
        try {
            if (!stop) {
                snakeController.step()
            }
        } catch (exception: SnakeDeadException) {
            stop = true
            snakeController.stop()
        }
    }

}
